server/server.go

```diff
@@ -263,6 +265,54 @@ func (r *rodeServer) ListOccurrences(ctx context.Context, occurrenceRequest *pb.
 	}, nil
 }
 
+func (r *rodeServer) ValidatePolicy(ctx context.Context, policy *pb.ValidatePolicyRequest) (*pb.ValidatePolicyResponse, error) {
+	log := r.logger.Named("ValidatePolicy")
+
+	if len(policy.Policy) == 0 {
+		return nil, createError(log, "empty policy passed in", nil)
+	}
+
+	// Generate the AST
+	mod, err := ast.ParseModule("validate_module", string(policy.Policy))
+	if err != nil {
+		log.Debug("failed to parse the policy", zap.Any("policy", err))
+		return &pb.ValidatePolicyResponse{
+			Policy:  policy.Policy,
+			Compile: false,
+			Errors:  []string{err.Error()},
+		}, nil
+	}
+
+	// Create a new compiler instance and compile the module. Should this be moved to initialize function?
+	c := ast.NewCompiler()
+
+	mods := map[string]*ast.Module{
+		"validate_module": mod,
+	}
+
+	if c.Compile(mods); c.Failed() {
+		log.Debug("compilation error", zap.Any("payload", c.Errors))
+		length := len(c.Errors)
+		errorsList := make([]string, length)
+
+		for i := range c.Errors {
+			errorsList = append(errorsList, c.Errors[i].Error())
+		}
+		return &pb.ValidatePolicyResponse{
```

Comment:
If this was a pure HTTP endpoint, I think this case and failing to parse would return a bad request (400) status, along with the errors. 

That might also remove the need for a compile flag.

Reply:
I debated on this myself for some time.
I always thought that if an endpoint behaves as it should, then it should return a 200. This is an endpoint that validates policy. It does that successfully as long as you pass in a nonempty policy.

The createpolicy endpoint is supposed to create a policy. If the policy you pass in doesn't compile, it returns a non 200 status code because it failed to do its job  i.e. create a policy.

That may be the wrong, but I wanted to get your opinion @alexashley

Reply:
My only concern is that I think users might expect a 200 as meaning that the policy is valid, without bothering to look at the response body or documentation. Usually if a client provides invalid data, that would be some form of a 4xx response. 

I don't feel too strongly about it either way -- there aren't any other users of this endpoint other than the UI and it's one of those RPCs that doesn't map in a straightforward way to a REST API.

Reply:
I don't either. I'm actually quite clueless here so I'll defer to you all. I'll merge this in so we can allow jazz to merge her changes. I can change this with the update policy pr

Reply:
I think you both make really good points. I don't feel particularly strong either way, but I agree with Alex that we might avoid some confusion switching to a 4xx status code since it's likely that there will be HTTP users of Rode.